assets/js/modules/analytics/account-create.js

```diff
@@ -0,0 +1,227 @@
+let timezoneData = false;
+
+const AccountCreate = () => {
+	const { createAccount } = dispatch( STORE_NAME );
+	const { siteName, siteURL, timezone: tz, errorcode } = global.googlesitekit.admin;
+	const [ error, setError ] = useState( errorcode );
+
+	const handleSubmit = useCallback( ( accountName, propertyName, profileName, timezone ) => {
+		trackEvent( 'analytics_setup', 'new_account_setup_clicked' );
+		createAccount( {
+			accountName,
+			propertyName,
+			profileName,
+			timezone,
+		} ).then( ( e ) => {
+			setIsSubmitting( false );
+			const { error: err } = e.payload;
+			if ( err ) {
+				setError( err.message ? err.message : __( 'Unknown error.', 'google-site-kit' ) );
+			}
+		} );
+	} );
+
+	// Build the timezone selector and cache it for re-renders.
+	const getTimezoneSelector = ( { timezone, setTimezone } ) => {
+		if ( timezoneData ) {
+			return timezoneData;
+		}
+		const { timezones } = global.googlesitekit.admin;
+
+		timezoneData = (
+			<Select
+				className="googlesitekit-analytics__select-timezone"
+				name="timezone"
+				style={ { minWidth: '240px' } /*todo: move to css */ }
+				enhanced
+				value={ timezone }
+				onChange={ ( e ) => {
+					setTimezone( e.target.value );
+				} }
+				label={ __( 'Timezone', 'google-site-kit' ) }
+				outlined
+			>
+				{ timezones && timezones
+					.map( ( aTimezone, index ) =>
+						<Option
+							key={ index }
+							value={ aTimezone.value }
+						>
+							{ aTimezone.name }
+						</Option>
+					) }
+			</Select>
+		);
+		return timezoneData;
+	};
+
+	const [ accountName, setAccountName ] = useState( siteName );
+	const [ propertyName, setPropertyName ] = useState( siteURL );
+	const [ profileName, setProfileName ] = useState( __( 'All website traffic', 'google-site-kit' ) );
+	const [ validationIssues, setValidationIssues ] = useState( {} );
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+	const [ timezone, setTimezone ] = useState( tz );
+
+	// Disable the submit button if there are validation errors, and while submission is in progress.
+	const buttonDisabled = validationIssues.accountName || validationIssues.propertyName || validationIssues.profileName || isSubmitting;
+
+	return (
+		<Fragment>
+			<div className="googlesitekit-setup-module">
+				<div className="mdc-layout-grid__inner">
+					<div className="mdc-layout-grid__cell mdc-layout-grid__cell--span-12">
+						<h2>
+							{ __( 'Create new Analytics account', 'google-site-kit' ) }
+						</h2>
+						{
+							error &&
+							<div className="error">
+								<p>
+									{ error }
+								</p>
+							</div>
+						}
+						<div>
+							{
+								isSubmitting
+									? <ProgressBar />
+									: <div>
+										<p>
+											{ __( 'Confirm your account details:', 'google-site-kit' ) }
+										</p>
+										<div className="googlesitekit-setup-module__inputs">
+											<div className="mdc-layout-grid__cell mdc-layout-grid__cell--span-2">
+												<TextField
```

Comment:
One initial feedback item for now: Let's implement separate components for individual pieces rather than creating one massive component like this one. See #1311 for inspiration, e.g. how it has `AccountSelect`, `PropertySelect` etc.

Reply:
Good suggestion, I'll work on that next.